qbft/messages.go

```diff
@@ -166,3 +166,56 @@ func Sign(msg *Message, operatorID types.OperatorID, operatorSigner types.Operat
 
 	return signedSSVMessage, nil
 }
+
+// ProcessingMessage stores the network-exchanged signed message and the decoded SignedMessage.SSVMessage.Data (qbft message)
+// The signed message is used at the qbft level since it's used for qbft justifications
+type ProcessingMessage struct {
+	SignedMessage *types.SignedSSVMessage
+	QBFTMessage   *Message
+}
+
+// NewProcessingMessage creates a ProcessingMessage with the decoded qbft message
+func NewProcessingMessage(signedMessage *types.SignedSSVMessage) (*ProcessingMessage, error) {
+	msg := &Message{}
+	err := msg.Decode(signedMessage.SSVMessage.Data)
+	if err != nil {
+		return nil, err
+	}
+	return &ProcessingMessage{
+		SignedMessage: signedMessage,
+		QBFTMessage:   msg,
+	}, nil
+}
+
+// Validate checks the signed message and qbft message validation
+func (msg *ProcessingMessage) Validate() error {
+
+	// Validate
+	if err := msg.SignedMessage.Validate(); err != nil {
+		return errors.Wrap(err, "invalid SignedSSVMessage")
+	}
+
+	// Check if the decoded message is correct
+	decodedMsg := &Message{}
+	err := decodedMsg.Decode(msg.SignedMessage.SSVMessage.Data)
```

Comment:
validation purpose is to validate message that comes from unknown actor on the network (we probably have duplication with p2p message validation)...

I don't think we need to decode all the time in spec